ProjFS.Mac/PrjFSKextTests/MockProc.cpp

```diff
@@ -0,0 +1,23 @@
+#include "MockProc.hpp"
+#include <string.h>
+
+int proc_pid(proc_t)
+{
+    return 1;
+}
+
+char* return_proc_name;
+
+void proc_name(int pid, char * buf, int size)
+{
+    if (return_proc_name != nullptr)
+    {
+        strcpy(buf, return_proc_name);
+    }
+}
+
+void SetProcName(char* procName)
+{
+    return_proc_name = strdup(procName);
```

Comment:
We should `free()` the previous memory before overwriting the pointer. Or just use a `std::string` for the variable holding the mock value and don't worry about memory.